src/main/java/tdl/client/queue/transport/RemoteBroker.java

```diff
@@ -1,75 +1,174 @@
+        client = createAWSClient(
+                serviceEndpoint,
+                config.getString("sqs.signingRegion"),
+                config.getString("sqs.accessKey"),
+                config.getString("sqs.secretKey")
+        );
 
         LoggerFactory.getLogger(RemoteBroker.class).debug("Connecting to the remote broker");
-        connection.start();
-        session = connection.createSession(false, Session.CLIENT_ACKNOWLEDGE);
 
-        String requestQueue = uniqueId + ".req";
-        messageConsumer = session.createConsumer(session.createQueue(requestQueue));
+        String requestQueue = uniqueId + "-req";
+        receiveMessageRequest = new ReceiveMessageRequest();
+        receiveMessageRequest.setMaxNumberOfMessages(MAX_NUMBER_OF_MESSAGES);
+        String queueUrl = String.format("%s/queue/%s", serviceEndpoint, requestQueue);
+        receiveMessageRequest.setQueueUrl(queueUrl);
+        receiveMessageRequest.setWaitTimeSeconds(MAX_AWS_WAIT);
+        receiveMessageRequest.setMessageAttributeNames(Arrays.asList(ATTRIBUTE_EVENT_NAME, ATTRIBUTE_EVENT_VERSION));
 
-        String responseQueue = uniqueId + ".resp";
-        messageProducer = session.createProducer(session.createQueue(responseQueue));
-        messageProducer.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+        deleteMessageBatchRequest.setQueueUrl(queueUrl);
 
-        this.requestTimeoutMillis = requestTimeoutMillis;
-        serializationProvider = new JsonRpcSerializationProvider();
+        String responseQueue = uniqueId + "-resp";
+        messageProducer = new SqsEventQueue(client, serviceEndpoint, responseQueue);
+
+        mapper = new ObjectMapper();
+
+        gson = new GsonBuilder()
+                .serializeNulls()
+                .create();
+
+        logToConsole("     RemoteBroker creation [end]");
     }
 
-    public Optional<Request> receive() throws BrokerCommunicationException {
-        try {
-            //Debt: We should have no timeout. This method could exit if we put a special close message in the queue
-            StringMessage messageText = new StringMessage(messageConsumer.receive(requestTimeoutMillis));
-            return serializationProvider.deserialize(messageText);
-        } catch (JMSException | DeserializationException e) {
-            throw new BrokerCommunicationException(e);
+    public List<Request> receive() throws BrokerCommunicationException {
+        logToConsole("     RemoteBroker receive [start]");
+            try {
+                List<Message> messages = client.receiveMessage(receiveMessageRequest).getMessages();
+                logToConsole("     RemoteBroker messages: " + messages);
+
+                List<Request> successfulMessages = processRequests(messages);
+
+                if (successfulMessages.size() > 0) {
+                    logToConsole("     RemoteBroker deleting read messages");
+                    deleteReadMessages(messages);
```

Comment:
It's a very bad idea to consume the messages before sending the response corresponding to that message.
This means that if the response fails, the message is consumed forever and the candidate will get a penalty. They will be very angry if they got a penalty due to network slowness and they will take it to us.

Reply:
This is because the test are cheating and asserting on the internals instead of the final state of the queue.
The failure scenarios are supposed to cover this.

Reply:
Hopefully resolved via b293e1326ef52bd4f205ea01645fb9dbfe2ce455